File: src/ServerPulse.Frontend/src/app/modules/authentication/domain/dtos/loginOAuthAccessTokenData.ts
import { AccessTokenData } from "../..";

export interface LoginOAuthAccessTokenData {
    accessToken: string;
    refreshToken: string;
    refreshTokenExpiryDate: Date;
}

export function mapLoginOAuthAccessTokenDataToAuthToken(response: LoginOAuthAccessTokenData): AccessTokenData {
    return {
        accessToken: response?.accessToken,
        refreshToken: response?.refreshToken,
        refreshTokenExpiryDate: new Date(response?.refreshTokenExpiryDate)
    }
}

File: src/ServerPulse.Backend/AuthenticationApi.Infrastructure/Dtos/Endpoints/OAuth/LoginOAuth/LoginOAuthAccessTokenData.cs
namespace AuthenticationApi.Infrastructure.Dtos.Endpoints.OAuth.LoginOAuth
{
    public class LoginOAuthAccessTokenData
    {
        public string AccessToken { get; set; } = string.Empty;
        public string RefreshToken { get; set; } = string.Empty;
        public DateTime? RefreshTokenExpiryDate { get; set; }
    }
}
